assets/model_monitoring/components/src/shared_utilities/io_utils.py

```diff
@@ -213,13 +175,15 @@ def save_spark_df_as_mltable(metrics_df, folder_path: str, file_system=None):
 
     retries = 0
     while True:
-        if _write_mltable_yaml(mltable_obj, folder_path, file_system):
+        if _write_mltable_yaml(mltable_obj, folder_path):
             break
         retries += 1
         if retries >= MAX_RETRY_COUNT:
             raise Exception("Failed to write mltable yaml file after multiple retries.")
         time.sleep(1)
 
+    metrics_df.write.mode("overwrite").parquet(base_path+"/data/")
```

Comment:
why removed the original try except logic?

Reply:
We don't need to catch the credential-less unavailable errors anymore. I flipped the logic order to write the MLTable yaml file first using StoreUrl so if credential-less doesn't work it will fail at that step. Now, when we get to the `spark.write` credential-less should work or it would have failed before.